week1/assignment/todo-loi/todo.rb

```diff
@@ -0,0 +1,123 @@
+    file = File.open(file_name)
+    data = file.read
+    items = []
+    data.lines.each do |line|
+      items << Item.new(line)
+    end
+    @list = List.new(items)
+  end
+
+  def seperate(title)
+    puts "\n"
+    puts ("-" * 75).colorize(:green)        
+    puts title.colorize(:green)
+    puts ("-" * 75).colorize(:green)    
+  end
+
+  def display
+    @list.all
+  end
+
+  def prompt
+    puts ("-" * 75).colorize(:green)
+    print "Please enter an option: ".colorize(:green)
+  end
+
+  def input(input)
+    input = input.downcase
+    case input 
+    when "display", "-d"
+      display
+    when "help", "-h"
+      help
+    when "add", "-a"
+      @list.add
+    when "remove", "-r"
+      @list.remove
+    when "mark", "-m"
+      @list.mark_complete
+    when "unmark", "-um"
+      @list.unmark_complete
+    when "colors"
+      show_colors      
+    else
+      puts "Sorry!".colorize(:light_red) +" I don't understand that command."
+      puts "Please try again or enter" + " -h".colorize(:light_red) + " or" + " help".colorize(:light_red) + " for more help"
+    end
+  end
+
+  def help 
+    puts "-h ".colorize(:light_red) + "or" + " help:".colorize(:light_red) + " Displays a list of all commands available."
+    puts "-d ".colorize(:light_red) + "or" + " display:".colorize(:light_red) +" Displays your todo list with all items."
+    puts "-a ".colorize(:light_red) + "or" + " add:".colorize(:light_red) + " Allows you to add new items to your todo list."
+    puts "-r ".colorize(:light_red) + "or" + " remove:".colorize(:light_red) + " Allows you to remove items from your todo list."
+    puts "-m ".colorize(:light_red) + "or" + " mark:".colorize(:light_red) + " Allows you to check items on the list as complete."
+    puts "-um ".colorize(:light_red) + "or" + " unmark:".colorize(:light_red) + " Allows you to uncheck items as complete on your todo list."
+  end
+
+  def show_colors
+    puts "black".colorize(:black)
+    puts "light_black".colorize(:light_black)
+    
+    puts "red".colorize(:red)
+    puts "light_red".colorize(:light_red)
+
+    puts "green".colorize(:green)
+    puts "light_green".colorize(:light_green)
+
+    puts "yellow".colorize(:yellow)    
+    puts "light_yellow".colorize(:light_yellow)
+
+    puts "blue".colorize(:blue)
+    puts "light_blue".colorize(:light_blue)
+
+    puts "magenta".colorize(:magenta)    
+    puts "light_magenta".colorize(:light_magenta)
+
+    puts "cyan".colorize(:cyan)    
+    puts "light_cyan".colorize(:light_cyan)
+
+    puts "white".colorize(:white)    
+    puts "light_white".colorize(:light_white)    
+
+    puts "default".colorize(:default)    
+  end  
+
+
+  def save
+    open('todo.data', 'w') { |f|
+    @list.items.each do |item|
+      f.puts item.current_status + " " + item.title
+    end
```

Comment:
nice! this works well and dont' change anything.
another way is to use `@list.items.map{...}.join("\n")` and then `f.puts` a single mapped result